### PR TITLE
Reorganize jenkins scripts

### PR DESCRIPTION
**Pull request by plally**

### CODE REVIEW

File: syntax_checker.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from pathlib import Path
+from subprocess import Popen, PIPE
+from threading import Thread
+
+# logging setup
+logger = logging.getLogger(__name__)
+coloredlogs.install(level='DEBUG', logger=logger, isatty=True, fmt='%(levelname)s %(message)s')
+
+
+class SyntaxChecker:
```
Comment (plally) [inline syntax_checker.py:14]: this name could probably use some improvement

Comment (sarahsturgeon) [reply syntax_checker.py:14]: I think it has to be pretty vague. Even vaguer than this, I think. Probably like `JobRunner` or something

File: syntax_checker.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+            p = Popen(command, stdout=PIPE, stderr=PIPE)
+            stdout, stderr = p.communicate()
+
+            for result in self.get_results(stdout, stderr):
+                self.results.append(result)
+
+
+    def get_results(self, stdout, stderr):
+        raise NotImplementedEror
```
Comment (sarahsturgeon) [inline syntax_checker.py:36]: What's your thinking with this?

Comment (plally) [reply syntax_checker.py:36]: the only places where all the build scripts really differ is when getting results from stdout/stderr
so I put that in its own function and it cant really be defined in the base class

**Comment by plally** [comment]
this just needs to be tested

**Comment by sarahsturgeon** [comment]
Tested linting and syntax checking of glua. Didn't try moonscript but I'll throw caution to the wind

**Pull request merged**